### DISCUSSION TITLE
External LED (WS2812x) not working

### DISCUSSION THREAD

**Discussion by frahemmi** [jomjol/AI-on-the-edge-device#1910] [General] [answered]
### The Problem

Hi, I have an issue that was already observed in the past and discussed here (https://github.com/jomjol/AI-on-the-edge-device/issues/682).
While I face exactly the same issue, none of the provided tipps solve my problem.
I try to use external LEDs (WS2812x type, 12 pieces) at GPIO12. I connected them according to the. schematic provided in the documentation.  I filled in all fields in the config, even the name field, but non of the LEDs is working. I tried some other WS2812x LEDs but same effect.
I enabled the DEBUG option for logging, but there is nothing related to my problem in the log file.
I tried the LEDS with a different board and some other software and found that the LEDs are working okay.

Is there a known issue (I was not able to find something here) or anything else that I can try?
Is it possible to use a different Pin?

### Version

Release: v13.0.8 (Commit: 04f2f23+)

### Logfile

```shell
[0d00h05m33s] 2023-01-23T17:43:06 <WRN> [OTA] Reboot in 5sec
[0d00h00m04s] 2023-01-23T16:43:19 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-01-23T16:43:19 <INF> [MAIN] ==================== Startup ====================
[0d00h00m04s] 2023-01-23T16:43:19 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-01-23T16:43:19 <INF> [MAIN] Branch: 'HEAD', Revision: 04f2f23+, Date/Time: 2022-12-19 20:41, Web UI: Release: v13.0.8 (Commit: 04f2f23+)
[0d00h00m04s] 2023-01-23T16:43:19 <INF> [MAIN] Reset reason: Via esp_restart
[0d00h00m08s] 2023-01-23T16:43:23 <INF> [MAIN] =================================================
[0d00h00m08s] 2023-01-23T16:43:23 <INF> [MAIN] ================== Main Started =================
[0d00h00m08s] 2023-01-23T16:43:23 <INF> [MAIN] =================================================
[0d00h00m08s] 2023-01-23T16:43:23 <INF> [MAIN] The device has 8 MBytes of PSRAM
[0d00h00m10s] 2023-01-23T16:43:25 <INF> [MAIN] Initialization completed successfully!
[0d00h00m14s] 2023-01-23T17:43:29 <INF> [SNTP] Time zone set to CET-1CEST,M3.5.0,M10.5.0/3
[0d00h00m14s] 2023-01-23T17:43:29 <DBG> [TFLITE SERVER] ----------------------------------------------------------------
[0d00h00m14s] 2023-01-23T17:43:29 <INF> [TFLITE SERVER] Round #1 started
```

### Expected Behavior

external LEDs should work on GPIO12

**Comment by caco3** [comment]
According to https://github.com/jomjol/AI-on-the-edge-device/issues/1402#issuecomment-1326133328 WS2812x is working!

Maybe @Ingo-Knito can give an additional hint?

**Comment by Ingo-Knito** [comment]
Did you see the arrow on the LED strips and solder the wires accordingly?

The correct connection is important, otherwise the serial control of the shift registers on each LED does not work and no LED lights up.

**Reply by frahemmi** [reply]
Hi.
First of all: Thanks for the immediate reply! Much appreciated. 
@Ingo-Knito Yes, I considered the direction. I also added the proposed resistor and capacitor. I also checked already if there is 5V on the LED Strip, which is okay.
![20230124_072453](https://user-images.githubusercontent.com/75826205/214227303-e929d602-4c3b-43dd-939e-876dfa2537ea.jpg)

**Comment by frahemmi** [comment]
Hi,
yes I checked between +5V and GND with a multimeter. 5.something volt were measured.
Power supply seems to be okay. My suspicion is, that I do not have a PWM signal on GPIO12, but I don’t know how to check this.
Do you know if there is a way to I use a different pin with same software version for external LED to try this?

Von: Ingo-Knito ***@***.***>
Antworten an: jomjol/AI-on-the-edge-device ***@***.***>
Datum: Dienstag, 24. Januar 2023 um 08:04
An: jomjol/AI-on-the-edge-device ***@***.***>
Cc: Frank Hemmersbach ***@***.***>, Author ***@***.***>
Betreff: Re: [jomjol/AI-on-the-edge-device] External LED (WS2812x) not working (Discussion #1910)


Have you checked the 5V against GND at the LEDs with a meter?

**Comment by Ingo-Knito** [comment]
It is not a PWM signal. It is a serial signal - like a shift register - to transfer the RGB values to each LED. Therefore you have to specify how many LEDs are included in the string.
A short signal interruption leads to the simultaneous takeover of the info of each LED. This only as info.

Do you have access to an oscilloscope? Try to check GPIO 12 and the LED.
The ESP32 should - if configured correctly - provide this signal at the PIN 12.

Does your config look like this?
<img width="450" alt="image" src="https://user-images.githubusercontent.com/39132657/214271239-633a4e9c-c3a9-4884-aee8-1c7db4c51675.png">

Additional check the config.ini with the file server. The section GPIO should look like this:
```
[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
IO12 = external-flash-ws281x disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812B
LEDNumbers = 12
```

**Reply by frahemmi** [reply]
Hi @Ingo-Knito ,
thanks for the additional details. this makes sense. 
Unfortunately I do nit have access to an oscilloscope.
Yes, my config looks like yours. Please see below:
<img width="311" alt="Snag_15e83c0" src="https://user-images.githubusercontent.com/75826205/214277160-afbeacbb-6d48-453d-9e01-2c841419377a.png">

<img width="220" alt="Snag_15eeba1" src="https://user-images.githubusercontent.com/75826205/214277249-29fac6e9-b03d-4ab6-b928-c8ca58a798e7.png">

**Comment by frahemmi** [comment, marked as answer]
Hi, 
here is an update: I solved the problem. 
After all the investigations there was the suspicion left, that something must be wrong with the hardware. I ordered a new ESP32 CAM module and tried again. This solved the problem.
Everything works as expected now!
Thanks @Ingo-Knito  for your efforts!

Regards
Frank
